### PR TITLE
Aux Mining Base Construction Console

### PR DESCRIPTION
**Pull request by GunHog**
- Adds the base construction console (code only)
- Adds the consoles to the maps (Box and Meta) and moves things around a little.

The base construction console is an advanced camera console, similar to the one employed in Xenobiology. It is locked into the aux mining base's building site and is used as a remote RCD for easy and fast customization of the mining base.

The console comes with a high-speed, large capacity internal RCD and the ability to build four tiny fans, for sane atmos management for up to four exits.

**NEW!** The base can now build up to four plasma cutter turrets for defense, calibrated to never target humans. Note that standing between them and a target is a very poor idea! Also make sure they have plenty of clearance, as windows and walls can block them.

I am adding this because I am disappointed in the lack of use for the base. A common complaint is that there is not enough time in the round to prepare a base and drop it. This is, hopefully, one step towards addressing that concern. The console is a way of allowing people to quickly and easily reshape and customize their base (It does NOT have to be a square!) to their liking. Engineering and Science teams can more quickly focus on setting up the interior how they please.

**Second and third sprites by @chanoc1!**

:cl: Gun Hog
add: Nanotrasen Engineering has devised a construction console to assist with building the Auxiliary Mining Base, usually located near a station's Arrivals hallway. A breakthrough in bluespace technology, this console employs an advanced internal Rapid Construction Device linked to a camera-assisted holocrane for rapid, remote construction!
/:cl:

image

### CODE REVIEW

**Comment by PKPenguin321** [comment]
like build mode but not

**Comment by optimumtact** [comment]
nobody uses the aux base because there's really no point tbh, there is already a base on lavaland and you can't really get anything by constructing another one

**Comment by FantasticFwoosh** [comment]
Base takes too long to build manually, and getting that double door right means blocking the entrance door to the actual building area awkwardly.

Having a 'build computer' attached to a camera in other places like the construction room (as a mini EVA, some materials to slot into the machine in the spare boxes) attached to the station camera grid so engineers can materialise in walls slowly or when there isn't a organic within 5 or so tiles would be nice.

Building your own build computer too would be the future, a bit like slime consoles.

**Comment by GunHog** [comment]
I have never used the Fulton stuff, explain it a little to me? Do you want the computer to spawn the beacons or the balloon thingies?

**Comment by FantasticFwoosh** [comment]
A return beacon would be nice for cargo to send supplies directly to the aux base (though it can just arrive in person using the landing craft) eh, nevermind.

A fulton beacon recieves the bag connected to it, so a miner would need to step outside with loaded bag full of ores and send it to a beacon in cargo. 

> Besides expanding or defining a camera network in the aux base would let you build outside its normal radius, for a module next door (or just a corridor to a pod nearby lazily as a living space)

**Comment by GunHog** [comment]
In a future PR, I would like to have a mining shuttle dock mapped to the station's aux base doc which becomes enabled once the mining shuttle beacon is activated. This will allow the mining shuttle to dock where the mining base used to live, thus resolving another concern where players returning from the base via the shuttle could only go to Cargo.

**Comment by GunHog** [comment]
Oh, regarding the fulton beacon: I would include one except that they need to be pre-linked in order to work. I am afraid to touch the maps more than necessary, fixing them is a huge pain. 

If there are any additional console features you would like to see considered, now is the best time to let me know!

**Comment by GunHog** [comment]
**New in this commit:**
- The base console may now build up to four plasma defense turrets,
calibrated to only attack hostile mobs, never humans.

- Added construction sounds for building things with the base console.

- Fixed a bug with tiny fans placement

The turrets tend to get taken out if more than one monster is attacking, but they can put up a somewhat acceptable performance.

**Comment by XDTM** [comment]
Does the aux base do damage when landing? I could see it as a megafauna-killing strat

**Comment by GunHog** [comment]
Go ahead. Try it. I dare you. Get a team and drop the Aux Base directly on top of a dragon. Please be sure to record it so I can laugh at you later.
## I HYPER CLOWN DARE YOU.

**Comment by Kromgar** [comment]
@GunHog Move the shuttle beacon to the mining base. I didn't even know it was a thing until recently

**Comment by GunHog** [comment]
I actually did put it inside the base now, next to the pillar.

**Comment by Kromgar** [comment]
@GunHog Pillar? What do you mean by pillar?

**Comment by GunHog** [comment]
The wall in the middle, I call it the "central pillar".

**Comment by Kromgar** [comment]
Oh @GunHog no i didn't mean in the auxillary base. But i guess that works. Most miners don't even know the beacon exists at all. I thought it would be a nice incentive to actually put down coordinates if the beacon was at the mining base. Most miners dont even carry the coordinate beacon for the base

**Comment by GunHog** [comment]
The base remote? It is at the top of every miner's locker. They do not carry them because miners deliberately ignore or drop the item. In my next PR, I would like to make it so that the base computer can choose a random point in Lavaland to drop if the miners never bother setting a drop point.

File: _maps/map_files/MetaStation/MetaStation.dmm [hunk 1]
```diff
@@ -1,5 +1,4 @@
 //MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE
-//Model dictionary trimmed on: 02-02-2017 02:51 (UTC)
```
Comment (optimumtact) [inline _maps/map_files/MetaStation/MetaStation.dmm]: you don't have to keep removing this comment

File: code/modules/mining/aux_base_camera.dm [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+	build_action.Grant(user)
+	airlock_mode_action.target = src
+	airlock_mode_action.Grant(user)
+	window_action.target = src
+	window_action.Grant(user)
+	fan_action.target = src
+	fan_action.Grant(user)
+	turret_action.target = src
+	turret_action.Grant(user)
+	eyeobj.invisibility = 0 //When the eye is in use, make it visible to players so they know when someone is building.
```
Comment (AnturK) [inline code/modules/mining/aux_base_camera.dm]: If you're doing this, give it IC sprite. Construction crane or something similar.

**Review by AnturK — APPROVED** [review]
Looks good, just some minor suggestion

File: code/modules/mining/aux_base_camera.dm (hunk 2 above)
Comment (RemieRichards) [inline code/modules/mining/aux_base_camera.dm]: little floating robot.

File: _maps/map_files/MetaStation/MetaStation.dmm (hunk 1 above)
Comment (GunHog) [inline _maps/map_files/MetaStation/MetaStation.dmm]: It happens automatically, I just forgot to revert it.

File: code/modules/mining/aux_base_camera.dm (hunk 2 above)
Comment (GunHog) [inline code/modules/mining/aux_base_camera.dm]: If I had such a sprite, I would happily do so!

Comment (GunHog) [inline code/modules/mining/aux_base_camera.dm]: I have an idea that might work splendidly! I will get on it soon!

Comment (chanoc1) [inline code/modules/mining/aux_base_camera.dm]: someone say sprites?

Comment (GunHog) [inline code/modules/mining/aux_base_camera.dm]: I have an excellent unused sprite from @WJohn I am going to try out!

Comment (chanoc1) [inline code/modules/mining/aux_base_camera.dm]: cool

**Comment by GunHog** [comment]
@RemieRichards, may I present to you the little robot concept:
screenshot 2017-02-10 16 18 47

This little sprite drone buddy is now what players will see when someone is using the building console! It even has movement sprites and animation! I extracted them from the mining borg PR where @WJohn made me these beautiful drone sprites that sadly went unmerged :cry:. Now, they have a new purpose and outlook on life!

**Comment by coiax** [comment]
I HATE TINY FANS SO MUCH

But this is a good idea.

**Pull request merged**